api_app/analyzers_manager/file_analyzers/filescan.py

```diff
@@ -0,0 +1,73 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+import time
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class FileScan(FileAnalyzer):
+    """Filescan Class"""
+
+    def set_params(self, params):
+        self.upload_file = params.get("upload_file", True)
+        self.session = requests.Session()
+        self.max_tries = 30
+        self.poll_distance = 10
+        self.request_url = "https://www.filescan.io/"
+
+    def run(self):
+        binary = self.read_file_bytes()
+        if not binary:
+            raise AnalyzerRunException("File is empty")
+        task_id = self.__filescan_request_scan(binary)
+        result = self.__poll_status(task_id)
+        return result
+
+    def __filescan_request_scan(self, binary) -> int:
+        logger.info(f"Requesting Scan for: ({self.filename}), {self.md5}")
+        name_to_send = self.filename if self.upload_file else self.md5
+        files = {"file": (name_to_send, binary)}
+        logger.info(f"Uploading for file analysis  of ({self.filename}), {self.md5}")
+        response = self.session.post(self.request_url + "api/scan/file", files=files)
+        if response.status_code != 200:
+            logger.info(f"Error: {response.status_code}")
+            raise AnalyzerRunException("Error Uploading File for Scan")
+        json_response = response.json()
+        task_id = json_response["flow_id"]
+        return task_id
+
+    def __poll_status(self, task_id: int) -> dict:
+        for chance in range(self.max_tries):
+            time.sleep(self.poll_distance)
+            url = self.request_url + "api/scan/" + str(task_id) + "/report"
```

Comment:
In the [FileScan API docs](https://www.filescan.io/api/docs#/scan/get_reports_api_scan__flow_id__report_get), it's mentioned that it is possible to retrieve more data from a report using filters.
![image](https://user-images.githubusercontent.com/16389167/144013411-d296a2ff-f28f-472f-bcf5-445a3fb3ec40.png)

Reply:
we are interested in almost all of them. For example, include atleast `wi:all`, `o:all`, `finalVerdict`, `dr:all`, `f:all`, `fd:all`, `general`.

Reply:
U mean something like this?
https://www.filescan.io/api/scan/619f888af36138de3f3c601c/report?filter=general&filter=subtaskReferences.state&filter=f%3AyaraMatches&filter=f%3AextractedFiles.yaraMatches&filter=f%3AextractedDomains.origin.type&filter=o%3Aresults&sorting=allSignalGroups%28description%3Aasc%2CaverageSignalStrength%3Adesc%29&sorting=allTags%28tag.name%3Aasc%29

Reply:
yes but use the filters I mentioned in the above comment.

Reply:
`requests.get(url, params={"filter": ["general",....]})` should probably work.

Reply:
Got It!!